test/unity/toolkits/drawing_classifier/test_dc_serialization.cxx

```diff
@@ -0,0 +1,182 @@
+      TS_ASSERT_EQUALS(convlayer.convolution().stride(0), 1);
+      TS_ASSERT_EQUALS(convlayer.convolution().stride(1), 1);
+      TS_ASSERT_EQUALS(convlayer.convolution().kernelsize(0), 3);
+      TS_ASSERT_EQUALS(convlayer.convolution().kernelsize(1), 3);
+      TS_ASSERT(convlayer.convolution().has_same());
+
+      const auto& relu_layer = nn.layers(layer_index + 1);
+      TS_ASSERT_EQUALS(relu_layer.name(),
+                       "drawing_relu" + std::to_string(x.first) + "_fwd");
+      TS_ASSERT(relu_layer.has_activation());
+      TS_ASSERT(relu_layer.activation().has_relu());
+
+      const auto& pool_layer = nn.layers(layer_index + 2);
+      TS_ASSERT(pool_layer.has_pooling());
+      TS_ASSERT_EQUALS(pool_layer.name(),
+                       "drawing_pool" + std::to_string(x.first) + "_fwd");
+      TS_ASSERT_EQUALS(pool_layer.pooling().kernelsize(0), 2);
+      TS_ASSERT_EQUALS(pool_layer.pooling().kernelsize(1), 2);
+      TS_ASSERT_EQUALS(pool_layer.pooling().stride(0), 2);
+      TS_ASSERT_EQUALS(pool_layer.pooling().stride(1), 2);
+      TS_ASSERT(pool_layer.pooling().has_valid());
+    }
+  }
+
+  unsigned layer_index = 10;
+  {
+    const auto& flatten_layer = nn.layers(layer_index);
+    TS_ASSERT(flatten_layer.has_flatten());
+    TS_ASSERT_EQUALS(flatten_layer.name(), "drawing_flatten0_fwd");
+  }
+
+  layer_index++;
+  {
+    const auto& dense_layer = nn.layers(layer_index);
+    TS_ASSERT(dense_layer.has_innerproduct());
+    TS_ASSERT_EQUALS(dense_layer.name(), "drawing_dense0_fwd");
+    TS_ASSERT_EQUALS(dense_layer.innerproduct().inputchannels(), 64 * 3 * 3);
+    TS_ASSERT_EQUALS(dense_layer.innerproduct().outputchannels(), 128);
+  }
+
+  layer_index++;
+  {
+    const auto& dense_layer = nn.layers(layer_index);
+    TS_ASSERT_EQUALS(dense_layer.name(), "drawing_dense1_fwd");
+    TS_ASSERT(dense_layer.has_innerproduct());
+    TS_ASSERT_EQUALS(dense_layer.innerproduct().inputchannels(), 128);
+    TS_ASSERT_EQUALS(dense_layer.innerproduct().outputchannels(), num_classes);
+  }
+
+  layer_index++;
+  TS_ASSERT(layer_index == 13);
+  {
+    const auto& softmax_layer = nn.layers(layer_index);
+    TS_ASSERT(softmax_layer.has_softmax());
+    TS_ASSERT_EQUALS(softmax_layer.output(0), target + "Probability");
+  }
+}
+
+BOOST_AUTO_TEST_CASE(test_export_coreml) {
+  // minimum startup code
+  const std::string target = "target";
+  const std::vector<std::string> features = {"0", "1"};
+  const std::vector<std::string> labels = {"0", "1"};
+
+  turi::drawing_classifier::drawing_classifier dc;
+  dc.add_or_update_state(
+      {{"target", target},
+       {"num_classes", labels.size()},
+       {"classes", flex_list(labels.begin(), labels.end())},
+       {"max_iterations", 300},
+       {"warm_start", false},
+       {"features", flex_list(features.begin(), features.end())}});
+
+  auto ml_model_wrapper = dc.export_to_coreml("", /* debug no throw */ true);
+  // TS_ASSERT(ml_model_wrapper != nullptr);
```

Comment:
Did you want to restore this assert? If not, let's remove it.

Reply:
Hah, let me restore it. Good catch!